txcache/selection.go

```diff
@@ -53,8 +56,40 @@ func selectTransactionsFromBunches(bunches []bunchOfTransactions, gasRequested u
 			break
 		}
 
-		accumulatedGas += gasLimit
-		selectedTransactions = append(selectedTransactions, item.transaction)
+		cache.askAboutAccountNonceIfNecessary(item)
+
+		isInitialGap := item.transactionIndex == 0 && item.senderNonceTold && nonce > item.senderNonce
+		if isInitialGap {
+			if logSelect.GetLevel() <= logger.LogTrace {
+				logSelect.Trace("TxCache.selectTransactionsFromBunches, initial gap",
+					"tx", item.transaction.TxHash,
+					"nonce", nonce,
+					"sender", item.transaction.Tx.GetSndAddr(),
+					"senderNonce", item.senderNonce,
+				)
+			}
+
+			// Item was popped from the heap, but not used downstream.
+			// Therefore, the sender is completely ignored in the current selection session.
```

Comment:
It makes sense to not select any transaction of a given sender, if we've detected an initial gap.